runsc/boot/controller.go

```diff
@@ -452,6 +456,9 @@ func (cm *containerManager) ExecuteAsync(args *control.ExecArgs, pid *int32) err
 // Checkpoint pauses a sandbox and saves its state.
 func (cm *containerManager) Checkpoint(o *control.SaveOpts, _ *struct{}) error {
 	log.Debugf("containerManager.Checkpoint")
+	cm.inCheckpointing.CompareAndSwap(0, -1)
+	defer cm.inCheckpointing.CompareAndSwap(-1, 0)
```

Comment:
This doesn't look right. Why do we use CompareAndSwap here? Can more than one checkpoint operation run concurrently? If the answer is yest, so we probably need to use cm.inCheckpointing.Add(1); defer cm.inCheckpointing.Add(-1).